File: quantasma-app/src/main/resources/frontend/src/app/backtest.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from "@angular/common/http";
import {Criterion} from "./shared/criterion.model";
import {map} from "rxjs/internal/operators";
import {Observable} from "rxjs/index";
import {Backtest} from "./shared/backtest.model";

@Injectable({
  providedIn: 'root'
})
export class BacktestService {

  constructor(private http: HttpClient) { }

  public all(): Observable<Backtest[]> {
    return <Observable<Backtest[]>> this.http.get("backtest/all");
  }

  public get(name: string): Observable<Backtest> {
    return <Observable<Backtest>> this.http.get("backtest/" + name);
  }

  public test(name: string, json: string) {
    return this.http.post("backtest/" + name, json);
  }

  public criterions(): Observable<Criterion[]> {
    return this.http.get("backtest/criterions").pipe(
      map(arr => (<string[]> arr).map(str => <Criterion>{name: str}))
    );
  }
}

File: quantasma-app/backend/src/main/java/quantasma/app/service/BacktestService.java
package quantasma.app.service;

import quantasma.app.model.BacktestRequest;
import quantasma.app.model.BacktestScenario;
import quantasma.core.analysis.BacktestResult;

import java.util.List;
import java.util.Set;

public interface BacktestService {

    List<BacktestScenario> all();

    BacktestScenario get(String name);

    List<BacktestResult> test(String name, BacktestRequest request);

    Set<String> criterions();
}
